crates/bevy_math/src/cubic_splines.rs

```diff
@@ -665,10 +655,22 @@ impl<P: Point> CubicSegment<P> {
         // These are the polynomial coefficients, computed by multiplying the characteristic
         // matrix by the point matrix.
         let coeff = [
-            p[0] * c0[0] + p[1] * c0[1] + p[2] * c0[2] + p[3] * c0[3],
-            p[0] * c1[0] + p[1] * c1[1] + p[2] * c1[2] + p[3] * c1[3],
-            p[0] * c2[0] + p[1] * c2[1] + p[2] * c2[2] + p[3] * c2[3],
-            p[0] * c3[0] + p[1] * c3[1] + p[2] * c3[2] + p[3] * c3[3],
+            P::point_add(
```

Comment:
If we added a `Point(pub T)` type, we could define its operations (`Add`, `Sub`, etc.) to defer to these `PointX` alternatives similar to [Wrapping](https://doc.rust-lang.org/std/num/struct.Wrapping.html), which would make these kinds of algorithms much easier to read.

Reply:
I don't think we even need the wrapper. We just need `trait LinearConexSpace: Add + Sub + ...` with documentation about when it should be implemented.

Reply:
I'm assuming the wrapper would be needed if the `PointAdd` implementation was different to the `Add` implementation. If the `PointX` traits aren't needed, then I agree that we could avoid the wrapper.

Reply:
I don't see why we would need two different addition operations. Connecting to your other comment, seems like `Add` should just implement the correct group for each coordinate in the first place.

Reply:
I believe there were concerns around the meaning behind the meaning of `Add` for colour types, which is why the `ComponentAdd` trait was added. I'm guessing there's a similar motivation for `PointAdd`.

Reply:
See Alice's comment about `LinearConvexSpace`. I will probably request changes if the current form is approved.